dotCMS/src/main/java/com/dotcms/webdav/FsDirectoryResource.java

```diff
@@ -0,0 +1,263 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package com.dotcms.webdav;
+
+import com.dotcms.repackage.com.bradmcevoy.http.*;
+import com.dotcms.repackage.com.bradmcevoy.http.exceptions.NotAuthorizedException;
+import com.dotmarketing.util.Logger;
+
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+
+/**
+ * Represents a directory in a physical file system.
+ *
+ */
+public class FsDirectoryResource extends FsResource implements MakeCollectionableResource, PutableResource, CopyableResource, DeletableResource, MoveableResource, PropFindableResource, LockingCollectionResource, GetableResource {
+
+
+    
+    private final FileContentService contentService;
+
+    public FsDirectoryResource(String host, FileSystemResourceFactory factory, File dir, FileContentService contentService) {
+        super(host, factory, dir);
+        this.contentService = contentService;
+        if (!dir.exists()) {
+            throw new IllegalArgumentException("Directory does not exist: " + dir.getAbsolutePath());
+        }
+        if (!dir.isDirectory()) {
+            throw new IllegalArgumentException("Is not a directory: " + dir.getAbsolutePath());
+        }
+    }
+
+    @Override
+    public CollectionResource createCollection(String name) {
+        File fnew = new File(file, name);
+        boolean ok = fnew.mkdir();
+        if (!ok) {
+            throw new RuntimeException("Failed to create: " + fnew.getAbsolutePath());

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)